ql/src/java/org/apache/hadoop/hive/ql/security/authorization/command/CommandAuthorizerV2.java

```diff
@@ -185,6 +189,38 @@ private static void addHivePrivObject(Entity privObject, Map<String, List<String
           tableName2Cols.get(Table.getCompleteName(table.getDbName(), table.getTableName()));
       hivePrivObject = new HivePrivilegeObject(privObjType, table.getDbName(), table.getTableName(),
           null, columns, actionType, null, null, table.getOwner(), table.getOwnerType());
+      if(table.getStorageHandler() != null){
+        //TODO: add hive privilege object for storage based handlers for create and alter table commands.
+        if(hiveOpType == HiveOperationType.CREATETABLE ||
+                hiveOpType == HiveOperationType.ALTERTABLE_PROPERTIES ||
+                hiveOpType == HiveOperationType.CREATETABLE_AS_SELECT){
+          String storageuri = null;
+          Map<String, String> tableProperties = new HashMap<>();
+          Configuration conf = new Configuration();
+          tableProperties.putAll(table.getSd().getSerdeInfo().getParameters());
+          tableProperties.putAll(table.getParameters());
+          try {
+            Method methodIsImplemented = table.getStorageHandler().getClass().getMethod("getURIForAuth", Map.class);
```

Comment:
same here. Wouldnt a check just to see if the class is implementing the new interface, being added, enough? instead of checking to see if the method is implemented?